micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -155,11 +155,11 @@ void checkAndBindMetrics(MeterRegistry registry) {
                 for (Meter other : registry.find(meterName).meters()) {
                     List<Tag> tags = other.getId().getTags();
                     // Only consider meters from the same client before filtering
-                    if (differentClient(tags)) break;
+                    if (differentClient(tags, metric.metricName().tags().get(CLIENT_ID_TAG_NAME))) continue;
                     if (tags.size() < meterTagsWithCommonTags.size()) registry.remove(other);
                     // Check if already exists
                     else if (tags.size() == meterTagsWithCommonTags.size())
-                        if (tags.equals(meterTagsWithCommonTags)) return;
+                        if (tags.containsAll(meterTagsWithCommonTags) && meterTagsWithCommonTags.containsAll(tags)) return;
```

Comment:
The `equals` method directly on the list objects would never return true, so an early return would never happen for this case, which I think isn't a big deal - it would just try registering a meter already registered, but this works around checking equality with this "clever hack".

Reply:
It seems that the latter condition can be omitted as its outer `if` statement checks if they have the same size. I created https://github.com/micrometer-metrics/micrometer/pull/2313.